pyquil/paulis.py

```diff
@@ -753,6 +759,8 @@ def exp_wrap(param):
             prog.inst(PHASE(-param * coeff, 0))
             prog.inst(X(0))
             prog.inst(PHASE(-param * coeff, 0))
+        elif is_zero(term):
+            prog.inst(I(0))
```

Comment:
this is a new behavior, especially since `I` imparts a delay when running on the hardware (I think)

Reply:
The desired behavior was really "do nothing". And since `PauliTerm("I", 0).program == Program()` and the fact that empty `Program`s can be run without errors, I'm replacing applying the identity to qubit 0 with literally doing nothing.

Reply:
Thanks!